synapseclient/client.py

```diff
@@ -3557,24 +3557,30 @@ def _updateEntity(self, entity, incrementVersion=True, versionLabel=None):
         Update an existing entity in Synapse.
 
         :param entity: A dictionary representing an Entity or a Synapse Entity object
+        :param incrementVersion: whether to increment the entity version (if Versionable)
+        :param versionLabel: a label for the entity version (if Versionable)
+
 
         :returns: A dictionary containing an Entity's properties
         """
 
         uri = '/entity/%s' % id_of(entity)
 
+        params = {}
         if is_versionable(entity):
-            if incrementVersion or versionLabel is not None:
-                uri += '/version'
+            if versionLabel:
+                entity['versionLabel'] = str(versionLabel)
+                incrementVersion = True
+
+            elif incrementVersion:
                 if 'versionNumber' in entity:
                     entity['versionNumber'] += 1
```

Comment:
Does setting this do anything? Can we just set label to current version + 1?
We don't seem to increment the version number if a versionLabel is provided since this is in an elif block

Reply:
If we don't need the version bump, then this logic can be simplified to deciding what label to use

Reply:
You're right, setting the entity version number doesn't have any affect, so we don't need to do that. I modified the conditionals accordingly.